File: app-java/src/main/java/com/java/bilibili/base/BaseActivity.java
package com.java.bilibili.base;


import android.app.Activity;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import com.java.bilibili.MainApplication;
import com.java.bilibili.manager.AppManager;

import butterknife.ButterKnife;
import butterknife.Unbinder;

public abstract class BaseActivity extends AppCompatActivity {
    protected Activity mActivity;
    private Unbinder mUnBinder;
    protected MainApplication app = MainApplication.getInstance();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayout());
        mUnBinder = ButterKnife.bind(this);
        mActivity = this;
        AppManager.getAppManager().addActivity(this);
        initEventAndData();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        AppManager.getAppManager().finishActivity(this);
        mUnBinder.unbind();
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

    protected abstract int getLayout();

    protected abstract void initEventAndData();
}

File: app-kotlin/src/main/java/com/kotlin/bilibili/base/BaseActivity.kt
package com.kotlin.bilibili.base

import android.app.Activity
import android.content.res.Configuration
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import com.kotlin.bilibili.MainApplication
import com.kotlin.bilibili.manager.AppManager


abstract class BaseActivity : AppCompatActivity() {
    lateinit var activity: Activity
    protected var app: MainApplication.Companion = MainApplication

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(layout())
        activity = this
        AppManager.appManager.addActivity(this)
        initData()
    }

    override fun onDestroy() {
        super.onDestroy()
        AppManager.appManager.finishActivity(this)
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
    }

    protected abstract fun layout(): Int

    protected abstract fun initData()
}
